### ISSUE TITLE
management console: TODO: Better 'loading' state styling

### ISSUE THREAD

**Issue by slimsag** [sourcegraph/sourcegraph#1822]
Currently it does not indicate well through the UX that the save operation is happening when you click Save (it only displays some "Saving..." text). A spinner of some sort would be better.

This is especially true because some connections to clusters can be a bit flaky, it is hard to know the save is still occurring.

**Comment by slimsag** [comment]
@attfarhan Would you be able to look into this since you have some spare cycles?

Basically, I am thinking instead of the `Saving...` text in the bottom left of the page (which is easy to miss) we could instead have a spinner and 'grey out' the actual text editor (e.g. maybe by overlaying a div with semi-transparent background); probably doesn't need design just needs to be more prominent somehow -- WDYT?

**Comment by chrismwendt** [comment]
How about just changing the button while saving? We could disable it, change the text to "Saving...", and put a loading spinner in or next to it.

I think we can get the loading spinner's CSS into the management console like this:

```scss
// cmd/management-console/web/src/CriticalConfigEditor.scss
@import '../node_modules/@sourcegraph/react-loading-spinner/lib/LoadingSpinner.css';
// ...
```

**Comment by slimsag** [comment]
That sounds good 👍 

However, I don't think we should import relative from `../node_modules`, instead we should `yarn add` that dependency inside `cmd/management-console/web`

**Comment by chrismwendt** [comment]
:+1: I added it in a WIP PR https://github.com/sourcegraph/sourcegraph/pull/2103 but ran into a problem right away. Let's discuss there.

**Comment by chrismwendt** [comment]
This won't get resolved in 3.0, moved to backlog.
